### PR TITLE
added priority percent to show policy map for iosxe

### PR DESCRIPTION
**Pull request by mc4leaf**
added priority percent to show policy map for iosxe.

### Description
https://github.com/CiscoTestAutomation/genieparser/issues/793

### CODE REVIEW

**Comment by SohanTirpude** [comment]
Hello @mc4leaf,

I have left few comments to improve your PR. Along with those, kindly add unit tests as well such as mentioned in this directory: https://github.com/CiscoTestAutomation/genieparser/tree/master/src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal

Kindly let me know if you have any questions.

Thank you.

File: src/genie/libs/parser/iosxe/show_policy_map.py [hunk 1]
```diff
@@ -2300,7 +2300,8 @@ class ShowPolicyMapSchema(MetaParser):
                     Any(): {
                         Optional('priority_level'): {
                             Any(): {
-                                'kbps': int}},
+                                'kbps': int,
```
Comment (SohanTirpude) [inline src/genie/libs/parser/iosxe/show_policy_map.py:2303]: Kindly revert back the changes to the original.

File: src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal/golden_output16_output.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+ Policy Map map1
+    Class cs1
+      priority 9 (%)
+    Class cs2
+      bandwidth 9 (%)
```
Comment (SohanTirpude) [inline src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal/golden_output16_output.txt:5]: Bandwidth value is not getting captured in your output, kindly work on regex p7 as well. This [regex](https://regex101.com/r/X2OXmT/1) may be suitable for this.

**Review by SohanTirpude — CHANGES_REQUESTED** [review]

File: src/genie/libs/parser/iosxe/show_policy_map.py (hunk 1 above)
Comment (mc4leaf) [reply src/genie/libs/parser/iosxe/show_policy_map.py]: corrected and restored.

File: src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal/golden_output16_expected.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+                "name": "cs2",
+                "bandwidth_percent": 9,
+                },
+            {
+                "name": "cs2",
+                "bandwidth_percent": 31,
+                }
+            ]
+        }
+    }
```
[the remark this answers is not in the document]
Comment (mc4leaf) [reply src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal/golden_output16_expected.py]: corrrected

File: changelog/undistributed/changelog_show_policy_map_iosxe_20231127192427.rst [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Only one changelog file per pull request. Combine these two templates where applicable.
+
```
[the remark this answers is not in the document]
Comment (mc4leaf) [reply changelog/undistributed/changelog_show_policy_map_iosxe_20231127192427.rst]: corrected

File: src/genie/libs/parser/iosxe/show_policy_map.py [hunk 5]
```diff
@@ -2525,6 +2527,10 @@ def cli(self, name='', output=None):
 
         # priority level 1
         p10_1 = re.compile(r'^priority +level +(?P<priority_levels>(\d+))$')
+        # priority 8 (%)
+
```
[the remark this answers is not in the document]
Comment (mc4leaf) [reply src/genie/libs/parser/iosxe/show_policy_map.py]: corrected

File: src/genie/libs/parser/iosxe/show_policy_map.py [hunk 6]
```diff
@@ -2300,7 +2300,8 @@ class ShowPolicyMapSchema(MetaParser):
                     Any(): {
                         Optional('priority_level'): {
                             Any(): {
-                                'kbps': int}},
+                                'kbps': int,
+                                'percent': int}},
```
[the remark this answers is not in the document]
Comment (mc4leaf) [reply src/genie/libs/parser/iosxe/show_policy_map.py]: corrected

File: src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal/golden_output16_output.txt (hunk 2 above)
Comment (mc4leaf) [reply src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal/golden_output16_output.txt]: updated output to focus on priority, previous bandwidth included as a mistake taken from working examples.

**Comment by mc4leaf** [comment]
Updates completed as requested, please review as nessesary.

**Review by mc4leaf — COMMENTED** [review]
comments actioned,

File: src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal/golden_output16_expected.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
+expected_output = {
+    'policy_map': {
+        'Policy-Map 1': {
+            'class': {
+                'Class-1': {
+                    'priority': True,
+                    'priority_percent': 18,
+                },
+                'Class-2': {
```
Comment (SohanTirpude) [inline src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal/golden_output16_expected.py:9]: Your expected output does not match with the output you shared in txt file. Kindly re-work on it.

Comment (SohanTirpude) [inline src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal/golden_output16_expected.py:9]: Your expected output does not match with the output you shared in txt file. Kindly re-work on it.

**Review by SohanTirpude — CHANGES_REQUESTED** [review]

File: src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal/golden_output16_expected.py (hunk 7 above)
Comment (mc4leaf) [reply src/genie/libs/parser/iosxe/tests/ShowPolicyMap/cli/equal/golden_output16_expected.py]: Corrected and tests now working as expected.

File: src/genie/libs/parser/iosxe/show_policy_map.py [hunk 8]
```diff
@@ -2843,6 +2847,13 @@ def cli(self, name='', output=None):
                 priority_level = int(m.groupdict()['priority_levels'])
                 class_map_dict['priority_levels'] = priority_level
 
+            # priority 9 (%)
+            m = p10_2.match(line)
+            if m:
+                class_map_dict['priority'] = True
+                priority_percent = int(m.groupdict()['priority_percent'])
+                class_map_dict['priority_percent'] = priority_percent
+
```
Comment (SohanTirpude) [inline src/genie/libs/parser/iosxe/show_policy_map.py:2856]: Kindly remove this blank line.

File: src/genie/libs/parser/iosxe/show_policy_map.py [hunk 9]
```diff
@@ -2843,6 +2847,13 @@ def cli(self, name='', output=None):
                 priority_level = int(m.groupdict()['priority_levels'])
                 class_map_dict['priority_levels'] = priority_level
```
Comment (SohanTirpude) [inline src/genie/libs/parser/iosxe/show_policy_map.py:2849]: Kindly add `continue` statement along with new line.

**Review by SohanTirpude — CHANGES_REQUESTED** [review]

File: src/genie/libs/parser/iosxe/show_policy_map.py (hunk 8 above)
Comment (mc4leaf) [reply src/genie/libs/parser/iosxe/show_policy_map.py]: done

File: src/genie/libs/parser/iosxe/show_policy_map.py (hunk 9 above)
Comment (mc4leaf) [reply src/genie/libs/parser/iosxe/show_policy_map.py:2850]: Done

**Comment by SohanTirpude** [comment]
Hello @mc4leaf, kindly pull the latest dev code.

**Comment by mc4leaf** [comment]
Master merged into dev branch and pushed.

**Review by SohanTirpude — APPROVED** [review]

**Pull request merged**